### PR TITLE
Adjust meson build system to not treat prefix as DESTDIR (#109)

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

File: data/meson.build [hunk 1]
```diff
@@ -24,9 +24,7 @@ con2.set('libexecdir',
 con2.set('bindir',
          join_paths(get_option('prefix'),
                     get_option('bindir')))
-con2.set('localstatedir',
-         join_paths(get_option('prefix'),
-                    get_option('localstatedir')))
+con2.set('localstatedir', join_paths(get_option('localstatedir')))
```
Comment (hughsie) [inline data/meson.build]: I don't think you need join_paths here.

File: data/meson.build [hunk 2]
```diff
@@ -14,7 +14,7 @@ install_data(['org.freedesktop.fwupd.xml'],
 )
 
 install_data(['90-fwupd-devices.rules'],
-  install_dir : join_paths(get_option('prefix'), 'lib', 'udev', 'rules.d')
+  install_dir : '/lib/udev/rules.d'
```
Comment (hughsie) [inline data/meson.build]: Is it usual to have a leading slash? I've not seen that in other meson projects.

Comment (superm1) [inline data/meson.build]: If it's not included then it starts to put stuff in prefix, so yes it's required.

File: data/meson.build (hunk 1 above)
Comment (superm1) [inline data/meson.build]: Right, will correct that.

**Comment by hughsie** [comment]
I've pulled in two hunks of this to master, but I've worked out why I'm having trouble with the rest; in Fedora we really do install files to `/usr/lib/systemd/system/` and `/usr/lib/udev/rules.d/` -- i.e. in the prefix. On Ubuntu/Debian, is `/lib` anything different to a symlink to `/usr/lib`?

**Comment by superm1** [comment]
With the systemd unit, the right thing to do should be to parse the pkg-config file for systemd.  It's properly indicated there.

Eg:
```
$ cat /usr/share/pkgconfig/systemd.pc
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

prefix=/usr
systemdutildir=/lib/systemd
systemdsystemunitdir=/lib/systemd/system
```

For the rules.d directory, I'm not sure where that's configurable.  It is not a symlink out to /usr/lib.  Is it a symlink on Fedora?

**Comment by hughsie** [comment]
Agreed, I think I've done that in 2d8a7958b2208ff4df8f93b0353ac99dc4519881 just now.

**Pull request closed without merging**